app/panel/components/BuildingBlocks/GhosteryFeature.jsx

```diff
@@ -14,83 +14,125 @@
-	 */
-	clickTrustButton() {
-		if (this.props.isInactive) {
-			return;
+		switch (type) {
+			case 'trust':
+				return (sitePolicy === WHITELISTED ?
+					t('summary_trust_site_active') :
+					t('summary_trust_site'));
+			case 'restrict':
+				return (sitePolicy === BLACKLISTED ?
+					t('summary_restrict_site_active') :
+					t('summary_restrict_site'));
+			default:
+				return 'Check button type you are passing to GhosteryFeature for typos and make sure it is being handled by getButtonText';
 		}
-		this.props.clickButton('trust');
 	}
 
-	/**
-	 * Handles the click event for the Custom Settings button
-	 */
-	clickCustomButton() {
-		if (this.props.isInactive) {
-			return;
+	_getTooltipText(sitePolicy, type) {
+		switch (type) {
+			case 'trust':
+				return (sitePolicy === WHITELISTED ?
+					t('tooltip_trust_on') :
+					t('tooltip_trust'));
+			case 'restrict':
+				return (sitePolicy === BLACKLISTED ?
+					t('tooltip_restrict_on') :
+					t('tooltip_restrict'));
+			default:
+				return 'Check button type you are passing to GhosteryFeature for typos and make sure it is being handled by getTooltipText';
 		}
-		this.props.clickButton('custom');
 	}
 
-	/**
-	 * Handles the click event for the Restrict Site button
-	 */
-	clickRestrictButton() {
-		if (this.props.isInactive) {
-			return;
+	_isFeatureActive(type, sitePolicy) {
+		switch (type) {
+			case 'trust':
+				return sitePolicy === WHITELISTED;
+			case 'restrict':
+				return sitePolicy === BLACKLISTED;
+			default:
+				return false;
 		}
-		this.props.clickButton('restrict');
+	}
+
+	render() {
+		const {
+			blockingPausedOrDisabled,
+			narrow,
+			short,
+			showText,
+			sitePolicy,
+			tooltipPosition,
+			type
+		} = this.props;
+
+		const active = this._isFeatureActive(type, sitePolicy);
+		// TODO Foundation dependency: button
+		const ghosteryFeatureClassNames = ClassNames(
+			'button',
+			'g-tooltip',
+			'GhosteryFeatureButton',
+			{
+				'GhosteryFeatureButton--normal': !narrow && !short,
+				'GhosteryFeatureButton--short': short,
+				'GhosteryFeatureButton--narrow': narrow,
+				'GhosteryFeatureButton--inactive': !active,
+				'GhosteryFeatureButton--active': active,
+				trust: type === 'trust',
+				restrict: type === 'restrict',
+				clickable: !blockingPausedOrDisabled,
+				'not-clickable': blockingPausedOrDisabled,
+			}
+		);
+
+		// TODO Foundation dependency: flex-container, align-center-middle
+		return (
+			<div className={ghosteryFeatureClassNames} onClick={this.handleClick}>
+				<span className="flex-container align-center-middle full-height">
+					<span className="GhosteryFeatureButton__text">
+						{this._getButtonText(sitePolicy, showText, type)}
+					</span>
+				</span>
+				<Tooltip body={this._getTooltipText(sitePolicy, type)} position={tooltipPosition} />
+			</div>
+		);
 	}
 
 	/**
 	 * React's required render function. Returns JSX
 	 * @return {JSX} JSX for rendering the Ghostery Features portion of the Summary View
 	 */
+	/*
```

Comment:
There is a lot of unused code at the bottom of this file.

Reply:
It is gone